### PR TITLE
azure-cli-extensions.ssh: init at 2.0.5

### PR DESCRIPTION
**Pull request by Gordon-BP**
## Description of changes

1. Added myself to the list of maintainers
2. Created a new python3Package,  [oschmod](https://pypi.org/project/oschmod/), which is a dependency for the ssh extension
3. Added the [Azure CLI SSH Extension](https://learn.microsoft.com/en-us/cli/azure/ssh?view=azure-cli-latest)

If this PR looks familiar, it is because it has taken me a few tries to get it right 🙈 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
  - The base azure-cli extension has its own testing procedure outlined [in the readme](https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/az/azure-cli/README.md). I ran both commands successfully:
```
gordy@nixos ~/nixpkgs$ nix build --impure --expr 'with (import ./. {}); azure-cli.withExtensions [ azure-cli.extensions.ssh ]'         
gordy@nixos ~/nixpkgs$ ./result/bin/az extension list                                          
[
  {
    "experimental": false,
    "extensionType": "whl",
    "name": "ssh",
    "path": "/nix/store/cx6radpc854089vrlzxaql3im24cs2wr-azure-cli-extensions/ssh",
    "preview": false,
    "version": "2.0.5"
  }
]
```
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Gordon-BP** [comment]
Hey @katexochen, after the dumpster fire that was my last PR, I've moved everything here. Please let me know if there is anything missing!

File: pkgs/by-name/az/azure-cli/python-packages.nix [hunk 1]
```diff
@@ -55,6 +55,8 @@ let
           msal-extensions
           msal
           msrestazure
+          oras
+          oschmod
```
Comment (katexochen) [inline pkgs/by-name/az/azure-cli/python-packages.nix:59]:
```suggestion
```
This shouldn't be needed.

**Review by katexochen — APPROVED** [review]

**Comment by katexochen** [comment]
Thanks for your contribution and welcome as maintainer! :)

File: pkgs/development/python-modules/oschmod/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "oschmod";
+  version = "0.3.12";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-vsmSFvMWFe5lOypch8rPtOS2GEwOn3HaGGMA2srpdPM=";
```
Comment (natsukium) [inline pkgs/development/python-modules/oschmod/default.nix:15]:
```suggestion
    hash = "sha256-vsmSFvMWFe5lOypch8rPtOS2GEwOn3HaGGMA2srpdPM=";
```

**Pull request merged**